lib/pavilion/plugins/commands/run.py

```diff
@@ -148,6 +149,11 @@ def run(self, pav_cfg, args, out_file=sys.stdout, err_file=sys.stderr):
         failed_build = None
         # Building any tests that specify that they should be built before
         for test in all_tests:
+            # make sure source_download_name is not set without source_location
```

Comment:
1. You're raising an exception of a type that won't be caught, which is a bug.
2. If the exception were caught and printed as an error, the user would have no idea where the problem is.
3. The PavTest object is responsible for making sure the test config is valid, so this code belongs there.